app/utils/lava_log_parser.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Based on log2html.py:
+# https://github.com/khilman/lab/blob/master/log2html.py
+#
+# error/warn formatting, css, colors etc. stolen from Olof Johansson <olof@lixom.net>
+#
+
+import argparse
+import json
+import os
+import re
+import sys
+import subprocess
+import yaml
+try:
+    import cStringIO as StringIO
+except ImportError:
+    import StringIO
+
+HTML_HEAD = """\
+<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN">
+<html>
+<head>
+  <title>Boot log: {board}</title>
+  <style type="text/css">
+  body {{ background-color: black; color: white; }}
+  pre warn {{ color: #F88017; }}
+  pre err {{ color: red; }}
+  pre pass {{ color: green; }}
+  pre pyboot {{ color: yellow; }}
+  pre offline {{ color: darkgray; }}
+  A:link {{text-decoration: none }}
+  A:visited {{text-decoration: none }}
+  A:active {{text-decoration: none }}
+  A:hover {{text-decoration: bold; color: red; }}
+  </style>
+</head>
+<body>
+<h1>Boot log: {board}</h1>
+"""
+
+def run(log, boot, txt, html):
+    boot_result = boot.get('boot_result', 'Unknown')
+    if boot_result == "PASS":
+        boot_result_html = "<pass>PASS</pass>"
+    elif boot_result == "FAIL":
+        boot_result_html = "<err>FAIL</err>"
+    else:
+        boot_result_html = "<warn>{}</warn>".format(boot_result)
+
+    formats = {
+        'warning': '<warn>{}</warn>\n',
+        'error': '<err>{}</err>\n',
+    }
+    numbers = {'warning': 0, 'error': 0}
+    log_buffer = StringIO.StringIO()
+
+    for line in log:
+        level = line['lvl']
+        fmt = formats.get(level, None)
+        if fmt:
+            log_buffer.write(fmt.format(line['msg']))
+            numbers[level] += 1
+        elif level == 'target':
+            msg = line['msg']
+            log_buffer.write(msg)
+            log_buffer.write('\n')
+            txt.write(msg)
+            txt.write('\n')
+
+    html.write(HTML_HEAD.format(board=boot['board']))
+    html.write("<font size=-2><pre>\n")
```

Comment:
Using the `<font>` tag should be avoided, it is an obsolete tag and it was deprecated. Browsers support for it can be dropped anytime.

Since it is only applied to the `<pre>` tag and there is a style block at the top, add something there: `pre { font-size: 80% }`